control/server.py

```diff
@@ -87,6 +82,8 @@ def __exit__(self, exc_type, exc_value, traceback):
     def serve(self):
         """Starts gateway server."""
 
+        self.logger.info(f"Starting gateway {self.config.gateway.name}")
```

Comment:
Not a big deal, but logger messages should be interpolated lazily:
```suggestion
    def serve(self):
        """Starts gateway server."""

        self.logger.info("Starting gateway %s", self.config.gateway.name)
```